src/audio/copier/copier.c

```diff
@@ -1662,8 +1662,14 @@ static uint64_t copier_get_processed_data(struct comp_dev *dev, uint32_t stream_
 	uint64_t ret = 0;
 
 	if (cd->endpoint_num && cd->bsource_buffer != input) {
-		if (stream_no < cd->endpoint_num)
-			ret = comp_get_total_data_processed(cd->endpoint[stream_no], 0, input);
+		if (stream_no < cd->endpoint_num) {
+			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw)
+				ret = input == (bool)dev->direction ?
```

Comment:
ok

Reply:
no, sorry, this is still wrong. A type-cast doesn't help. This line is comparing apples and oranges. It's like comparing weight with colour. `input` is a boolean variable, it can be `true` or `false` for whatever condition it is meant for. `dev->direction` is a stream direction, something like `playback` or `capture`. You cannot compare them. What are you trying to check here? Maybe something like
```
ret = (input && dev->direction == SOF_IPC_STREAM_CAPTURE) ? ... : ...;
```
or similar? But that is probably wrong, just guessing. What are you checking here?